front: improve railway manager integration

SarahBellaha opened the pull request:
Closes [#1247](https://github.com/osrd-project/osrd-confidential/issues/1247)

:warning:  Needs to be merged after [this PR](https://github.com/OpenRailAssociation/osrd/pull/14283)

JulesPeignier committed:
railway_manager_interface: change openapi to allow substitute train, anterior and posterior train, add demand category, hazardous materials and request identifier

Signed-off-by: Jules Peignier <jules.peignier@gmail.com>

SarahBellaha committed:
front: add new fields in the railway manager modal

Signed-off-by: SarahBellaha <sarah.bellaha@sncf.fr>

SarahBellaha committed:
front: include request identifier in success message for last minute requests

Signed-off-by: SarahBellaha <sarah.bellaha@sncf.fr>

emersion commented on front/src/modules/SimulationReportSheet/utils/formatSimulationReportSheet.ts:
```diff
@@ -393,35 +392,16 @@ export const transformStepsToApiFormat = (
     path_date: similarTrains[0].start_time
-      ? new Date(similarTrains[0].start_time).toLocaleDateString(locale)
-      : new Date().toLocaleDateString(locale),
+      ? new Date(similarTrains[0].start_time).toISOString()
+      : new Date().toISOString(),
```
Thank you for fixing this! \o/

Closes https://github.com/OpenRailAssociation/osrd/issues/14373

SarahBellaha committed:
front: use ISO string instead of locale date formatting

- We use ISO (UTC) date format to prevent issues related to user locale or timezone differences

Signed-off-by: SarahBellaha <sarah.bellaha@sncf.fr>

Uriel-Sautron commented on front/src/applications/stdcm/styles/_railwayManagerModal.scss:
This comes from a previous PR, but the border-radius in the `.pdf-download` class should be `4px`, not `8px`.

Uriel-Sautron reviewed (approved):
LGTM for code but i haven't tested it by sending a form to GESICO.

SarahBellaha committed:
front: update PDF download icon styling

Signed-off-by: SarahBellaha <sarah.bellaha@sncf.fr>

SharglutDev commented on front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx:
The bullet points are not the good ones it seems

App : 

<img width="259" height="124" alt="Image" src="https://github.com/user-attachments/assets/f34c9ddb-6cb3-4d4f-84d8-0d535b080347" />

Mockup : 

<img width="385" height="148" alt="Image" src="https://github.com/user-attachments/assets/d109a709-e961-49ed-a4da-3a21a623ad6a" />

SarahBellaha replied:
It's the default styling of the `<li>` bullet points, i can add a `:before` element to have smaller ones :)

SharglutDev replied:
I was more thinking of the "bullet point" character :)

SharglutDev commented on front/src/applications/stdcm/styles/_railwayManagerModal.scss:
```diff
@@ -181,29 +188,80 @@
+        }
+
+        p:first-child {
+          font-size: 16px;
```
font-size should be in rem

SharglutDev commented on front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx:
```diff
@@ -134,6 +155,13 @@ const SendToRailwayManagerModal = ({
     { value: 'Autre', label: t('modal.other') },
   ];
 
+  const trainCategoryOptions = [
+    DEFAULT_TRAIN_CATEGORY,
+    { value: 'A', label: 'A' },
+    { value: 'B', label: 'B' },
+    { value: 'C', label: 'C' },
+  ];
```
This can be moved outisde of the component

SarahBellaha replied:
Actually it can't since we have the `DEFAULT_TRAIN_CATEGORY` that contains a translated string in it : 
`const DEFAULT_TRAIN_CATEGORY = { value: '', label: t('modal.noCategory') };`

SharglutDev replied:
Oh right, my mistake

SharglutDev commented on front/public/locales/en/stdcm.json:
```diff
@@ -156,24 +156,34 @@
-        "verifyParameters": "Here are the query parameters and simulation that will be sent to GESICO. Please take the time to check them"
+        "trainCategory": "Train category",
+        "trainToBeSubstituted": "Train to be substituted",
+        "verifyParameters": "Here are the query parameters and simulation that will be sent to GESICO. Please take the time to check them",
```
nit
```suggestion
        "verifyParameters": "Here are the query parameters and simulation that will be sent to GESICO, Please take the time to check them.",
```

SharglutDev commented on front/public/locales/fr/stdcm.json:
```diff
@@ -156,24 +156,34 @@
+        "trainToBeSubstituted": "Sillon à substituer",
+        "verifyParameters": "Voici les paramètres de requête et la simulation qui seront transmis à GESICO, prenez le temps de les vérifier.",
+        "warnings": {
+          "hazardousMaterialsBold": "Les restrictions des convois de matières dangereuses ne sont pas pris en compte par l’algorithme LMR.",
```
Small diff compared to the mockup
```suggestion
          "hazardousMaterialsBold": "Les restrictions des convois porteurs de matières dangereuses ne sont pas prises en compte par l’algorithme LMR.",
```

SharglutDev commented on front/public/locales/fr/stdcm.json:
```diff
@@ -156,24 +156,34 @@
+        "verifyParameters": "Voici les paramètres de requête et la simulation qui seront transmis à GESICO, prenez le temps de les vérifier.",
+        "warnings": {
+          "hazardousMaterialsBold": "Les restrictions des convois de matières dangereuses ne sont pas pris en compte par l’algorithme LMR.",
+          "hazardousMaterialsNormal": "La simulation sera vérifiée par un horairiste lors du traitement du sillon, et pourrait être irrecevable."
```
Has this part changed compared to the mockup ?
```suggestion
          "hazardousMaterialsNormal": "La faisabilité de la demande sera étudiée par un horairiste lors du traitement du sillon, et pourrait être irrecevable."
```

SharglutDev commented on front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx:
```diff
@@ -412,6 +460,23 @@ const SendToRailwayManagerModal = ({
+                required
+              />
+              <div id="hazardous-materials">
+                <Checkbox
+                  label={t('modal.hazardousMaterials')}
+                  checked={isHazardousMaterials}
+                  onChange={() => setIsHazardousMaterials(!isHazardousMaterials)}
+                />
```
Probably an ui issue but the alignment between the checkbox and the label doesn't match the mockup

App : 
<img width="622" height="64" alt="Image" src="https://github.com/user-attachments/assets/55c0c6d6-068f-40cc-9dd3-1377e7ad29d7" />

Mockup: 
<img width="461" height="56" alt="Image" src="https://github.com/user-attachments/assets/32eb63bb-b9b6-4766-beed-12a8be27e434" />

SarahBellaha replied:
Yes, it comes from the checkbox styling :/

SharglutDev replied:
Would you mind opening a bug report for that ?

SarahBellaha replied:
Fixed in [that PR](https://github.com/OpenRailAssociation/osrd/pull/14634) :)

SharglutDev commented on front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx:
```diff
@@ -421,6 +486,28 @@ const SendToRailwayManagerModal = ({
+                }
+                label={t('modal.trainToBeSubstituted')}
+              />
+              <DatePicker
```
The width of the picker is a bit weird

<img width="1334" height="752" alt="Image" src="https://github.com/user-attachments/assets/2054d103-b2d3-4fb9-9446-b9f68024fd7b" />

SarahBellaha replied:
I agree. This is another ui issue 😅​

SharglutDev replied:
Same as above then :) But it really doesn't look good, we should handle this asap.

SarahBellaha replied:
Issue fixed in the same PR as the checkbox :arrow_up:

SharglutDev commented on front/src/applications/stdcm/styles/_railwayManagerModal.scss:
Seems like the color on the mockup is now `info60`

SharglutDev commented on front/src/applications/stdcm/styles/_railwayManagerModal.scss:
```diff
@@ -181,29 +188,80 @@
+        p {
+          min-height: 24px;
+          color: var(--warning60);
+          font-family: 'IBM Plex Sans';
```
Do we need this ?

SarahBellaha replied:
We indeed don't need to specify the font-family, thanks :)

SharglutDev commented on front/src/applications/stdcm/styles/_railwayManagerModal.scss:
```diff
@@ -181,29 +188,80 @@
+    }
+    .warning-section {
+      background-color: var(--ambientA15);
+      padding: 0 58px 44px 54px;
```
On the mockup the gap below the warning section is 124px but currently it's more like 156px~.

I know the gesico modal global architecture doesn't come from this PR but it's pretty hard to compare it to the mockup with all these paddings/margins distributed among multiple components :/

SarahBellaha replied:
The difference comes from the `margin-bottom: 32px` applied to all the `<section>` l.31.
Fixed :)

SharglutDev commented on front/src/modules/SimulationReportSheet/utils/formatSimulationReportSheet.ts:
```diff
@@ -390,16 +390,13 @@ export const transformStepsToApiFormat = (
     path_date: similarTrains[0].start_time
-      ? new Date(similarTrains[0].start_time).toLocaleDateString(locale)
-      : new Date().toLocaleDateString(locale),
+      ? new Date(similarTrains[0].start_time).toISOString()
```
Would you mind to explain in the commit description what this change fixes ?

SharglutDev reviewed (commented):
Thank you for the PR !

Ideally for next time, since this repo is public and confidential is note, it would be better to create an issue resuming the description and the ACs of the original one :)

Also would you mind rebasing your PR with dev ? A lot has changed in editoast and it's easier not to have to re-build containers when switching to this branch 🙂 (did it myselft since there isn't any conflicts but still a good practice to rebase a branch regularly)

Morgane-SJK commented on front/src/common/api/osrdRailwayManagerApi.ts:
```diff
@@ -448,7 +449,11 @@ export type SimulationReport = {
+  substitute_train?: SubstituteTrain | null;
   course_type: CourseType;
   statistical_category: string;
+  hazardous_materials?: boolean;
```
Why not  `hazardous_materials?: boolean | null;` ?

JulesPeignier replied:
` hazardous_materials` should not be optional, `front/src/common/api/osrdRailwayManagerApi.ts` is generated automatically from the OpenAPI, I think it's because I forgot to add 
```yaml
required:
- hazardous_materials
```

JulesPeignier replied:
Actually, we could also leave it optional and let the railway manager handle the behaviour, I don't have a strong opinion on this.

Morgane-SJK replied:
Ok! I also don't have a strong opinion about it but if you keep it like this, you should adapt the type of `hazardous_materials` in the class `SimulationReport` in `railway_manager_interface/osrd_railway_manager_interface/models.py`. Actually you have:
```
hazardous_materials: Annotated[bool | None, Field(title="Hazardous Materials")] = (
        None
    )
```

JulesPeignier replied:
`railway_manager_interface/osrd_railway_manager_interface/models.py` is also generated from the OpenAPI, I will discuss this with @SarahBellaha and make the necessary changes accordingly !

JulesPeignier replied:
done

Morgane-SJK commented:
Can you update the PR description with the steps to test your PR ? 🙏

SharglutDev reviewed (approved):
Lgtm (not tested to send to gesico). You can close the last comments when fixed.

Morgane-SJK commented:
> Can you update the PR description with the steps to test your PR ? 🙏

I could test thanks to [this README](https://gitlab-repo-res.apps.eul.sncf.fr/dsir/groupedxs-dsir/04735/osrd-data/-/merge_requests/483) ✅

<img width="1262" height="631" alt="Capture-GESICO" src="https://github.com/user-attachments/assets/3efb8b56-004d-4bf6-85a5-e27b4c401949" />
